README.md

```diff
@@ -106,38 +106,63 @@ For project using Scala 2.12
 
 ### Initial info file generation example
 
-Atum provides a method for initial creation of info files from a Spark dataframe. It can be used as is or can
+Atum provides helper methods for initial creation of info files from a Spark dataframe. It can be used as is or can
 serve as a reference implementation for calculating control measurements.
 
-The method accepts some metadata as parameters. In addition it accepts the list of fields for which control
+The `ControlMeasureBuilder.builder` accepts some metadata via optional setters. In addition it accepts the list of fields for which control
 measurements should be generated. Depending on the data type of a field the method will generate a different
 control measurement. For numeric types it will generate **controlType.absAggregatedTotal**, e.g. **SUM(ABS(X))**. 
 For non-numeric types it will generate **controlType.HashCrc32** e.g. **SUM(CRC32(x))**. Non-primitive data types 
 are not supoprted.   
 
 ```scala
-    val dataSourceName = "Source Application"
-    val inputPath = "data/input"
-    val batchDate = "10-10-2017"
-    val batchVersion = 1
+import java.net.URI
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import za.co.absa.atum.model.ControlMeasure
+import za.co.absa.atum.utils.controlmeasure.{ControlMeasureBuilder, ControlMeasureUtils}
+
+val dataSourceName = "Source Application"
+val inputPath = "/path/to/source"
+val batchDate = "15-10-2017"
+val batchVersion = 1
+
+val spark = SparkSession.builder()
+  .appName("An info file creation job")
+  .getOrCreate()
+
+val df: DataFrame = spark
+  .read
+  .format("csv").option("header", "true") // adjust to your data source format
+  .load(inputPath)
+val aggregateColumns = List("employeeId", "address", "dealId") // these columns must exist in the `df`
+
+// builder-like fluent API to construct a ControlMeasureBuilder and yield the `controlMeasure` with `build`
+val controlMeasure: ControlMeasure =
+  ControlMeasureBuilder.builder(df, aggregateColumns)
+    .withInputPath(inputPath)
+    .withSourceApplication(dataSourceName)
+    .withReportDate(batchDate)
+    .withReportVersion(batchVersion)
+    .build
+
+// convert to JSON using .asJson | asJsonPretty
+println("Generated control measure is: " + controlMeasure.asJson)
+
+// write to HDFS
+{
```

Comment:
the block here and below is used to scope otherwise ambiguous implicit `FileSystem` val.

Reply:
I would split the example into 3. Creation, write to HDFS, write to S3. And in the last two, you reference the `controlMeasure` val

Or explain the scoping in the md itself.

Reply:
OK, split into multiple examples. Explaining the scoping could be done (I really thought about it to keep the example intact), but it would break focus off the main effort too much for my linking.